reV/losses/power_curve.py

```diff
@@ -505,10 +502,153 @@ def target(self):
 
     @property
     def transformation(self):
-        """PowerCurveTransformation: Requested power curve transformation."""
+        """PowerCurveTransformation: Power curve transformation."""
         return TRANSFORMATIONS[self._transformation_name]
 
 
+class PowerCurveWindResource:
+    """Wind resource data for calculating power curve shift."""
+
+    def __init__(self, temperature, pressure, wind_speed):
+        """Power Curve Wind Resource.
+
+        Parameters
+        ----------
+        temperature : array_like
+            An iterable representing the temperatures at a single site
+            (in C). Must be the same length as the `pressure` and
+            `wind_speed` inputs.
+        pressure : array_like
+            An iterable representing the pressures at a single site
+            (in ATM). Must be the same length as the `temperature` and
+            `wind_speed` inputs.
+        wind_speed : array_like
+            An iterable representing the wind speeds at a single site
+            (in m/s). Must be the same length as the `temperature` and
+            `pressure` inputs.
+        """
+        self._temperatures = np.array(temperature)
+        self._pressures = np.array(pressure)
+        self._wind_speeds = np.array(wind_speed)
+        self.wind_speed_weights = None
+
+    def wind_resource_for_site(self):
+        """Extract scaled wind speeds at the resource site.
+
+        Get the wind speeds for this site, accounting for the scaling
+        done in SAM [1]_ based on air pressure [2]_. These wind speeds
+        can then be used to sample the power curve and obtain generation
+        values.
+
+        Returns
+        -------
+        array-like
+            Array of scaled wind speeds.
+
+        References
+        ----------
+        .. [1] Scaling done in SAM: https://tinyurl.com/2uzjawpe
+        .. [2] SAM Wind Power Reference Manual for explanations on
+           generation and air density calculations (pp. 18):
+           https://tinyurl.com/2p8fjba6
+
+        """
+
+        pressures_pascal = self._pressures * 101325.027383  # originally in atm

```

Comment:
Worth doing a simple check to see if p is in atm or Pa? WTK is in Pa. I know docstring says pressure should be in ATM but still...

Reply:
I would love this actually. Is there a good way to distinguish the unit type from the value alone? Just check for values over 1e6?

Reply:
Yeah i typically just set a threshold... Pa should be >1e4 and atm should be less than 2. These are generous but you can get some low pressures in very mountainous regions and whatnot.